src/main/telemetry/smartport.c

```diff
@@ -198,19 +262,29 @@ static void smartPortSendByte(uint8_t c, uint16_t *crcp)
     *crcp = crc;
 }
 
-static void smartPortSendPackage(uint16_t id, uint32_t val)
+static void smartPortSendPackageEx(uint8_t primId, uint8_t* data)
 {
     uint16_t crc = 0;
-    smartPortSendByte(FSSP_DATA_FRAME, &crc);
+    smartPortSendByte(primId, &crc);
+    for(uint8_t i=0;i<6;i++) {
```

Comment:
In CF I usually use unsigned when upper bound is unsigned (ARRAYLEN/sizeof) to avoid warning.

in C signed overflow is undefined, so compiler may generate `for` code as if overflow can't happen, possibly generating a bit better code ..